jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/list/AbstractMultiGridView.java

```diff
@@ -539,12 +539,14 @@ public ColumnSortList reloadColumnSortList() {
         if (gridPreferencesStore != null) {
             GridSortedColumnPreference gridSortedColumnPreference = gridPreferencesStore.getGridSortedColumnPreference();
             //Avoid duplicating the call push method of ColumnSortList when catch ColumnSortedEvent
-            //If repeating the call push method, there will throw 'IndexOutOfBoundsException: Row index: 0, Row size: 1'
             if (gridSortedColumnPreference != null && columnSortList.size() <= 1) {
                 Optional<ColumnMeta<T>> optional = getListGrid().getColumnMetaList().stream().filter(
                         tColumnMeta -> tColumnMeta.getColumn().getDataStoreName().equals(gridSortedColumnPreference.getDataStoreName()))
                         .findFirst();
                 if (optional.isPresent()) {
+                    if (columnSortList.get(0).getColumn().getDataStoreName().equals(gridSortedColumnPreference.getDataStoreName())) {
```

Comment:
What exactly are we doing here?

Reply:
This situation is difficult to describe, I try to describe it.
For Task List page, it have a default sort column createOn, and the save sort column is createOn, when you click createOn, here will push a createOn into columnSortList,(but the createOn Column's ascending is not same ), In the case, columnSortList's size is 1 ,so saved sort column will push int columnSortList.but push of columnSortList beehive is "If the column already exists, it will be removed from its current position and placed at the start of the list.", the columnSortList's size always is 1, so it never show correct sort column.

Reply:
I also try to fix it.PR:https://github.com/gwtproject/gwt/pull/9661.
and create a issue to discuss it.https://github.com/gwtproject/gwt/issues/9660
Suggestion is try to find work around way.

Reply:
Could we add a test case for the issue we're dealing with?

Reply:
Ok I will try to add some test for it.

Reply:
The test maybe is useless.  I try to describe it. I can see the remove logic in here https://github.com/gwtproject/gwt/blob/master/user/src/com/google/gwt/user/cellview/client/ColumnSortList.java#L208. "if (curInfo.getColumn() == column) ", so    we can not simulate our's situation. This is my test code , but it can not work.
https://github.com/bxf12315/gwt/blob/902de3d0210bfb95c4790a1fb10fa3b2d8aad104/user/test/com/google/gwt/user/cellview/client/ColumnSortListTest.java#L316

Reply:
butif it generate js code, it is working fine.

Reply:
Ok, thanks for looking into that.